dask_expr/_expr.py

```diff
@@ -1093,7 +1093,19 @@ def _task(self, index: int):
         return self.value
 
 
-class Blockwise(Expr):
+class BlockwiseOverlapping(Expr):
+    """Super class that operates like a blockwise op
+
+    We require information from neighboring partitions, so we can't prune partitions
+    before lowering, but the spec is the same as for Blockwise ops, we don't reoder
+    things, alignment stays consistent, ...
+
+    """
+
+    pass
+
+
```

Comment:
Why does this exist?  Is there some shared structure between these two that we need to handle?

My intuition here is just to have 

```python
class Blockwise(Expr):
    ...

class MapOverlap(Expr):
    ...
```

At least from this PR it doesn't seem like we're using the intermediate classes.

Reply:
My recommendation would be to wait until we have a demonstrable need, and even in that case to consider using `(Blockwise, MapOverlap)` instead if possible.